### PR TITLE
Enum for abstainVote

### LINKED ISSUE #230: Enum for abstainVote
**Is your feature request related to a problem? Please describe.**
- [ ] Before you work on this issue, please refer to: https://github.com/Mafia-Hands/Mafia/issues/178
- [ ] Before you work on this issue, please refer to: https://github.com/Mafia-Hands/Mafia/issues/178

Create an enum for "abstrain Vote", so that we do not just pass around a literal string.

**Describe the solution you'd like**
Enum representing abstrain Vote.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### Feature Approval:
- [x]  Feature approved by: @MKTPR

### PR DESCRIPTION
**Pull request by lhsunyc**
# Description
This PR fixes #230 

Create the abstain vote enum and extract it to the common folder and apply changes to relevant files.
# Demo
please refer to files changed

# Type of change:
* Bug fix (Non-breaking change which fixes a bug)

### CODE REVIEW

File: backend/domain/DTO/response/ListVoteDTO.js
Comment (JChamberlain28) [inline backend/domain/DTO/response/ListVoteDTO.js]: why is the name here 'evotes' but in other files, it is 'votes'. Also, a more descriptive name for this variable might be useful. What is it representing? 'votes' is semi ambiguous.

File: backend/domain/VoteHandler.js
Comment (JChamberlain28) [inline backend/domain/VoteHandler.js]: more desriptive name.

**Review by JChamberlain28 — CHANGES_REQUESTED** [review]
Consider these changes. Otherwise, it resolves the issue. Tested working on my computer.

File: backend/domain/DTO/response/ListVoteDTO.js
Comment (milkyway404) [reply backend/domain/DTO/response/ListVoteDTO.js]: I agree. Perhaps VoteType?

**Review by yzho976 — DISMISSED** [review]
These changes looks good to me. One thing Im not sure is why the code for TrialStateChangeEvents.js has been deleted and added again. But that's acceptable.

**Review by JChamberlain28 — DISMISSED** [review]
Thanks for adding the changes.
LGTM

Before merging you will need to ensure all tests pass though. There are 2 back end test failures.

File: backend/Events/GameStateEvents/TrialStateChangeEvents.js
Comment (ambiyu) [inline backend/Events/GameStateEvents/TrialStateChangeEvents.js]: Not sure why this was changed to `isAlive`? Please change it back to `status`:
```
       room.getPlayerByNickname(playerChosen).status = PlayerStatus.KILLED_BY_TOWN;
```

**Review by ambiyu — CHANGES_REQUESTED** [review]
Thanks for this. Resolving the requested change should fix the tests failing. Please also change all instances of voteType to VoteType (uppercase V) to keep enum naming consistent.

File: backend/Events/GameStateEvents/TrialStateChangeEvents.js
Comment (lhsunyc) [reply backend/Events/GameStateEvents/TrialStateChangeEvents.js]: The problem solved, thanks andy!

**Review by yzho976 — APPROVED** [review]
This looks good for me now.

**Review by ambiyu — APPROVED** [review]
LGTM now. Thanks for the changes

**Review by khal745 — CHANGES_REQUESTED** [review]
Looking good on the backend, but I think some frontend changes are required given we are using this new enum. Have a look at #291 where the `Table.jsx` and `useGameState.js` file required some small updates so they pass around the newly changed info. Because its the same string ('no Confidence') used in frontend compared to the enum value, I think the game should work, but passing the literal string in frontend doesn't solve the original issue. We could merge this if we need to though.

**Review by khal745 — APPROVED** [review]
For now we can merge and make a follow up PR to address the frontend issues.

**Pull request merged**
